exchanges/btcmarkets/btcmarkets_websocket.go

```diff
@@ -203,11 +203,16 @@ func (b *BTCMarkets) wsHandleData(respRaw []byte) error {
 		}
 
 		side := order.Buy
-		if t.Side == "Ask" {
+		switch {
+		case t.Side.IsLong():
+			// Nothing to do
+		case t.Side.IsShort():
 			side = order.Sell
+		default:
+			return fmt.Errorf("%w: `%s`", order.ErrSideIsInvalid, t.Side)
```

Comment:
🚧 🧪 And I guess now I realise what my original problem was:
There was a input path that would not hit the right code path.
Now we have the correct lines, but I don't think we have coverage of them yet.
Suggest we want to make test BUY, SELL, ASK, BID do the right thing in terms of what the trade we send down the DataHandler looks like, that INVALID hits a json unmarshal error, and  that ANY hits this line

Reply:
As mentioned, we'll want to use `FixtureToDataHandlerWithErrors` to get at the errors thrown straight out, require the correct len, and then check they match what we expected.
Reminder: `Invalid` will get caught by json, and `Any` would hit this line.